client/src/main.tsx

```diff
@@ -3,18 +3,21 @@ import ReactDOM from "react-dom/client";
 
 import { Login, Register, Loading, NotFound, EditProfile, Search } from "./components/pages";
 
-import { createBrowserRouter, RouterProvider, redirect, Outlet } from "react-router-dom";
+import { createBrowserRouter, RouterProvider, defer } from "react-router-dom";
 
 import { ChakraProvider } from "@chakra-ui/react";
 import { userProfileSchema } from "./schemas.ts";
+import { useProfileStore } from "./store.ts";
+import { getToken } from "./token.ts";
 
-const router = createBrowserRouter([
-  {
-    path: "/",
-    loader: async () => {
-      const token = localStorage.getItem("token") ?? null;
-
+const implicitLogin = async () => {
+  return new Promise((resolve, reject) => {
+    setTimeout(async () => {
```

Comment:
`setTimeout` is great for testing, but we probably shouldn't include it into the codebase